numpy/core/src/multiarray/ctors.c

```diff
@@ -1324,7 +1324,7 @@ _array_from_array_like(PyObject *op,
      *      this should be changed!
      */
     if (!writeable && tmp == Py_NotImplemented) {
-        tmp = PyArray_FromArrayAttr(op, requested_dtype, context);
+        tmp = PyArray_FromArrayAttr(op, requested_dtype, context); // Assume: array was copied.
```

Comment:
This seems like a no to me.  I expect even pandas breaks this and returns views (although they may be discussing to change it).  Or things like `xarray` probably return views?  Without an argument that this is correct, I have to assume it is not correct.

Reply:
You removed the comment, but the behaviour remains identical?
```
class MyArray:
    def __array__(self):
        return np.arange(10)
```
is not rejected.  Do we really want to live with that relaxed interpretation here?  For the buffer protocol and array-struct/interface I can see it, but not for `__array__`.